components/toast/Toast.js

```diff
@@ -76,40 +69,22 @@ class Toast extends PureComponent {
   };
 
   render() {
-    const { active, children, className, label, processing } = this.props;
-
-    const toast = (
-      <Transition in={active} timeout={{ enter: 0, exit: 1000 }}>
-        {state => {
-          if (state === 'exited') {
-            return null;
-          }
+    const { children, className, label, processing } = this.props;
 
-          const classNames = cx(
-            theme['toast'],
-            {
-              [theme['is-entering']]: state === 'entering',
-              [theme['is-entered']]: state === 'entered',
-              [theme['is-exiting']]: state === 'exiting',
-            },
-            className,
-          );
+    const classNames = cx(theme['toast'], className);
 
-          return (
-            <div data-teamleader-ui="toast" className={classNames}>
-              {processing && <LoadingSpinner className={theme['spinner']} color="white" />}
-              <TextBody className={theme['label']} color="white">
-                {label}
-                {children}
-              </TextBody>
-              {this.renderCustomAction() || this.renderCustomLink() || this.renderCloseButton()}
-            </div>
-          );
-        }}
-      </Transition>
+    return (
+      <div data-teamleader-ui="toast">
+        <div className={classNames}>
```

Comment:
Q: Just wondering now, is this second div still necessary? Isn't it possible to put `className={classNames}` on the div above it and get rid of this one?

Reply:
Good question. I came to doing this as follows:

How the transition works now when a toast disappears is as follows: in the first phase it goes to the right and fades out at the same time. With a small delay, the 2nd phase starts in which I set the toasts max-height to 0 so we can drop the remaining toast stack nice and gently. However, margins and paddings were still left, so when the toast leaves the DOM, it still had a height of +- 30px due to padding and bottom-margin still being applied, resulting in a very snappy animation and a jump. I tried to fix this by animating the top and bottom padding, but this was very laggy and jumpy as was the margin animation as well if applied at all. To overcome this I added another component around it without any margin or padding so it would not be affected by the laggy animations from the padding and margin.  

Thus: as we don't want any spacing on that outer div to keep control over animations, and any other styles would be redundant there, I don't think it makes sense to apply classNames on the outer div.

Reply:
Fair enough 👍 

 :ok: by @InstaK on 03/07/2018 13:42:18